src/core/server/saved_objects/migrations/core/index_migrator.ts

```diff
@@ -171,6 +172,31 @@ async function deleteIndexTemplates({ client, log, obsoleteIndexTemplatePattern
   return Promise.all(templateNames.map((name) => client.indices.deleteTemplate({ name: name! })));
 }
 
+/**
+ * Delete saved objects by type. This is used to remove saved object types that
+ * are not compatible with the current version of OpenSearch Dashboards.
```

Comment:
We are not actually adding logic to determine which saved object types are not compatible, right? We are deleting configured `typesToDelete`.

can we change the description to avoid confusion? maybe - `If migrations.delete.types is specified, this will delete any saved object type that match it`

Reply:
updated